### ISSUE TITLE
Jinja For Loop Breaking Fix Command When In CTE

### ISSUE THREAD

**Issue by tac14yo** [sqlfluff/sqlfluff#4667]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running sqlfluff fix on a model that is inside of a CTE that also has a Jinja For/In/Endfor loop we end up getting the "Loop limit on fixes reached"

We are able to override this issue by warpping the For Loop portion in a "-- noqa: disable=LT02"  

### Expected Behaviour

To be able to complete the fix task as there are fixable violations

Ideal situation would be able to fix all issues; including what is inside the For Loop

In previous version (1.4.5) this completed with no issues but seemed to ignore what was inside the For Loop

### Observed Behaviour

When running a similar model (lets call it sqlfluff_test) :

```sql
{% set test_field = ["a", "b", "c"] %}

with foo as (
    select * from vee
),

final as (

select
column_a,
column_b,
column_c,
column_d
from foo
where column_a = 'test'
and column_b is null
    {%- for test_field in test_field %}
    and ( lower({{ test_field }}) not like '%test%'
            or {{ test_field }} is null)
    {%- endfor %}   
        and ( lower(column_d) not like '%test%'
            or lower(column_d) not like '%last%'
                or column_d is null)
)

select * from final
```

We get the following response:

```
==== finding fixable violations ====
WARNING    Loop limit on fixes reached [10].
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [22 unfixable linting violations found]
```

### How to reproduce

To make sure it isn't a setting we ran the following on the above file:

```
sqlfluff fix models\sqlfluff_test.sql --ignore-local-config --dialect redshift
```

If we ignore the LT02 rule then it seems to work; BUT that is because 

```
sqlfluff fix models\sqlfluff_test.sql --ignore-local-config --dialect redshift --exclude-rules LT02 
```

We get the same results even if using dbt as the templater:

```
sqlfluff fix models\sqlfluff_test.sql --ignore-local-config --dialect redshift --templater dbt 

==== finding fixable violations ====
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
WARNING    Loop limit on fixes reached [10]. 
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [22 unfixable linting violations found]
```

### Dialect

redshift

### Version

sqlfluff, version 2.0.2
Python 3.9.16

sqlfluff templater dbt, version 2.0.2
DBT:
Core:
  - installed: 1.4.5
  - latest:    1.4.5 - Up to date!

Plugins:
  - postgres: 1.4.5 - Up to date!
  - redshift: 1.4.0 - Up to date!

### Configuration

This issue happend with default config

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
This was fixed between 2.0.2 and 2.0.3. Can you try again on 2.0.3.

I'm not sure exactly which PR fixed it, I'm going to guess that it was #4640, but I've confirmed this is fixed on 2.0.3.

On that basis I'm going to close this. Do reopen it if you still experience issues with this query on 2.0.3.

**Comment by tac14yo** [comment]
The new version no longer throws a "limit on fixes reached" and does actually fix, however now it is cutting off part of the statement and causing a parsing error because it is losing some of the statement.

As this is now technically a different issues; should this remain closed an I open a new one?
